services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js

```diff
@@ -0,0 +1,200 @@
+      key: "category",
+      filters: categoryFilters,
+      onFilter: (value, record) => record.category.includes(value),
+      sorter: (a, b) => (a.category > b.category ? -1 : 1),
+      render: (text) => <div title="Category">{text}</div>,
+    },
+    {
+      title: "Description",
+      dataIndex: "description",
+      key: "description",
+      sorter: (a, b) => (a.description > b.description ? -1 : 1),
+      render: (text) => <div title="Proponent Description">{text}</div>,
+    },
+    {
+      title: "Upload Date/Time",
+      dataIndex: "upload_date",
+      key: "upload_date",
+      sorter: (a, b) => (moment(a.upload_date) > moment(b.upload_date) ? -1 : 1),
+      render: (text, record) => (
+        <div title="Due">{formatDateTime(record.upload_date) || "Pending"}</div>
+      ),
+    },
+  ];
+};
+
+const transfromDocuments = (
+  documents,
+  now_application_guid,
+  noticeOfWorkApplicationDocumentTypeOptionsHash
+) =>
+  documents.map((document) => ({
+    key: document.id,
+    now_application_guid,
+    filename: document.mine_document.document_name || Strings.EMPTY_FIELD,
+    document_manager_guid: document.mine_document.document_manager_guid,
+    upload_date: document.mine_document.upload_date,
+    category:
+      (noticeOfWorkApplicationDocumentTypeOptionsHash &&
+        noticeOfWorkApplicationDocumentTypeOptionsHash[
+          document.now_application_document_type_code
+        ]) ||
+      Strings.EMPTY_FIELD,
+    description: document.description || Strings.EMPTY_FIELD,
+  }));
+
+const handleAddDocument = (closeDocumentModal, addDocument) => (values) => {
+  const document = {
+    now_application_document_type_code: values.now_application_document_type_code,
+    description: values.description,
+    is_final_package: values.is_final_package,
+    mine_document: {
+      document_manager_guid: values.document_manager_guid,
+      document_name: values.document_name,
+      mine_guid: values.mine_guid,
+    },
+  };
+  addDocument(FORM.EDIT_NOTICE_OF_WORK, "documents", document);
+  closeDocumentModal();
+};
+
+const openAddDocumentModal = (
+  event,
+  openDocumentModal,
+  closeDocumentModal,
+  addDocument,
+  now_application_guid,
+  mine_guid
+) => {
+  event.preventDefault();
+  openDocumentModal({
+    props: {
+      onSubmit: debounce(handleAddDocument(closeDocumentModal, addDocument), 2000),
+      title: `Add Notice of Work document`,
+      now_application_guid,
+      mine_guid,
+    },
+    content: modalConfig.EDIT_NOTICE_OF_WORK_DOCUMENT,
+  });
+};
+
+const propTypes = {
+  openModal: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  now_application_guid: PropTypes.string.isRequired,
+  mine_guid: PropTypes.string.isRequired,
+  documents: PropTypes.arrayOf(PropTypes.any).isRequired,
+  noticeOfWorkApplicationDocumentTypeOptionsHash: PropTypes.objectOf(PropTypes.any).isRequired,
+  arrayPush: PropTypes.func.isRequired,
+  isViewMode: PropTypes.bool.isRequired,
+};
+
```

Comment:
Unimportant change, but for clarity, can you move this to the top of the file.

Reply:
I thought that typically we have been putting function definitions that are not part of the component (stateless functions) above PropTypes in other files, but I see it's inconstant in different places. I'll move it :)